publisher/main.go

```diff
@@ -2,56 +2,86 @@ package main
 
 import (
 	"flag"
-	"log"
 	"net/http"
 	_ "net/http/pprof"
+	"os"
+	"os/signal"
+	"syscall"
 	"time"
 
-	"github.com/coreos/go-etcd/etcd"
-	"github.com/fsouza/go-dockerclient"
+	"github.com/Sirupsen/logrus"
 
 	"github.com/deis/deis/publisher/server"
 )
 
-const (
-	defaultRefreshTime time.Duration = 10 * time.Second
-	defaultEtcdTTL     time.Duration = defaultRefreshTime * 2
-	defaultHost                      = "127.0.0.1"
-	defaultDockerHost                = "unix:///var/run/docker.sock"
-	defaultEtcdHost                  = "127.0.0.1"
-	defaultEtcdPort                  = "4001"
-	defaultLogLevel                  = "error"
-)
-
 var (
-	refreshDuration = flag.Duration("refresh-duration", defaultRefreshTime, "The time to wait between etcd refreshes.")
-	etcdTTL         = flag.Duration("etcd-ttl", defaultEtcdTTL, "The TTL for all of the keys in etcd.")
-	host            = flag.String("host", defaultHost, "The host where the publisher is running.")
-	dockerHost      = flag.String("docker-host", defaultDockerHost, "The host where to find docker.")
-	etcdHost        = flag.String("etcd-host", defaultEtcdHost, "The etcd host.")
-	etcdPort        = flag.String("etcd-port", defaultEtcdPort, "The etcd port.")
-	logLevel        = flag.String("log-level", defaultLogLevel, "Acceptable values: error, debug")
+	bindAddr   string
+	interval   time.Duration
+	publishTTL time.Duration
+	host       string
+	dockerAddr string
+	etcdAddr   string
+	log        = logrus.New()
+	logLevel   string
 )
 
+func init() {
+	flag.StringVar(&bindAddr, "bind-addr", "localhost:6060", "address to listen for incoming HTTP requests")
```

Comment:
I don't really like the `init()` here, it just adds more burden on the binary boot. Why not simple have them all vars? like

```
var (
  bindAddr = flag.String("bind-addr", "localhost:6060", "address to listen for incoming HTTP requests")
)
```